src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,55 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
+
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_ELEMENTS = 10;
+    private int size = 0;
+    private K[] keyArray;
+    private V[] valueArray;
+
+    public StorageImpl() {
+        keyArray = (K[]) new Object[MAX_ARRAY_ELEMENTS];
+        valueArray = (V[]) new Object[MAX_ARRAY_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == 0) {
+            keyArray[size] = key;
+            valueArray[size] = value;
+            size++;
+        }
+        for (int i = 0; i < size; i++) {
+            if ((key != null && key.equals(keyArray[i]))
+                    || (key == keyArray[i])) {
+                valueArray[i] = value;
+                return;
+            }
+        }
+        if (size < MAX_ARRAY_ELEMENTS) {
+            keyArray[size] = key;
+            valueArray[size] = value;
+            size++;
+        } else {
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
            if ((key != null && key.equals(keyArray[i]))
                    || (key == keyArray[i])) {
                valueArray[i] = value;
                return;
            }
        }
        if (size < MAX_ARRAY_ELEMENTS) {
            keyArray[size] = key;
            valueArray[size] = value;
            size++;
        } else {
```
it will work without the first if